src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -321,6 +355,24 @@ public Activity AddTag(string key, string? value)
             return this;
         }
 
+        /// <summary>
+        /// Add <see cref="ActivityLink" /> object to the <see cref="Links" /> list.
+        /// </summary>
+        /// <param name="link"> object of <see cref="ActivityLink"/> to add to the attached links list.</param>
+        /// <returns>'this' for convenient chaining</returns>
+        public Activity AddLink(ActivityLink link)
```

Comment:
OpenTelemetry does not allow to add links after Span has started (links affect sampling decision). I suggest we do not allow it either in Activity

Reply:
good point, I'll apply your suggestion.

Reply:
I have addressed this issue. please look at the source updates.